feat: add support for span logs

ahmadshaheer opened the pull request:
## 📄 Summary
This PR adds support for displaying related signals (logs) for the selected span in trace details v2, improving trace-to-logs correlation. Users can now view contextual logs directly from span details, with related log lines highlighted and explained through an info icon. Clicking on log lines redirects to the logs explorer with trace and span filters applied.
  
<!-- Describe the purpose of the PR in a few sentences. What does it fix/add/update? -->

---

## ✅ Changes

- [ ] Feature: Brief description
- [ ] Bug fix: Brief description

---

## 🏷️ Required: Add Relevant Labels

> ⚠️ **Manually add appropriate labels in the PR sidebar**  
Please select one or more labels (as applicable):

ex:

- `frontend`
- `backend`
- `devops`
- `bug`
- `enhancement`
- `ui`
- `test`

---

## 👥 Reviewers

> Tag the relevant teams for review:

- frontend / backend / devops

---

## 🧪 How to Test

<!-- Describe how reviewers can test this PR -->
  1. Navigate to any trace detail page in SigNoz
  2. Click on any span to open the span details drawer
  3. Verify the "Logs" button is available in the related signals section
  4. Click the "Logs" button and verify:
    - A drawer opens showing related logs
    - Logs are filtered by the trace_id
    - Log lines related to the selected span are visually highlighted
    - An info icon explains why certain log lines are highlighted
  5. Click on any highlighted log line and verify:
    - You are redirected to the logs explorer
    - Both trace_id and span_id filters are properly applied
    - The selected log entry is visible in the explorer
  6. Test the feature with different spans to ensure proper filtering and highlighting

---

## 🔍 Related Issues

<!-- Reference any related issues (e.g. Fixes #123, Closes #456) -->
Closes https://github.com/SigNoz/signoz/issues/8387

---

## 📸 Screenshots / Screen Recording (if applicable / mandatory for UI related changes)


https://github.com/user-attachments/assets/412f2f8a-ee83-4502-b664-dab47ef36e4c




<!-- Add screenshots or GIFs to help visualize changes -->

---

## 📋 Checklist

- [ ] Dev Review
- [ ] Test cases added (Unit/ Integration / E2E)
- [ ] Manually tested the changes


---

## 👀 Notes for Reviewers

<!-- Anything reviewers should keep in mind while reviewing -->

<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds support for displaying and interacting with span-related logs in trace details, enhancing trace-to-log correlation.
> 
>   - **Behavior**:
>     - Adds span log support in `SpanDetailsDrawer.tsx`, allowing users to view logs related to a selected span.
>     - Clicking on log lines redirects to logs explorer with trace and span filters applied.
>   - **Components**:
>     - Introduces `SpanLogs.tsx` and `SpanRelatedSignals.tsx` for displaying and managing span-related logs.
>     - Updates `RawLogView` to handle log highlighting and click events.
>   - **Styles**:
>     - Adds styles in `SpanLogs.styles.scss` and `SpanRelatedSignals.styles.scss` for new components.
>   - **Tests**:
>     - Adds tests in `SpanDetailsDrawer.test.tsx` to verify log display and navigation behavior.
>   - **Misc**:
>     - Updates `reactQueryKeys.ts` with new keys for span logs.
>     - Adds `@signozhq/button` to `package.json` dependencies.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for 92d055d39a3dd71581be3b4992031d0658dd8088. You can [customize](https://app.ellipsis.dev/SigNoz/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

aks07 reviewed (dismissed):
LGTM

ahmadshaheer commented:
> * please use query range v5.
> * fetch logs for specific trace_id, span_id
> * then fetch the logs for trace_id with the  id filters from above
> 
>  Screen.Recording.2025-08-27.at.9.40.51.AM.mov 
> .

Done, @nityanandagohain

nityanandagohain reviewed (changes requested):
* Please use query range v5
* To load newer older logs, i.e `id >` please use [{key: {name: "timestamp"}, direction: "asc"}] . Check how it is done in logs context.
* When I click on explore button it uses `IN` operator. Why is that the case and `=` is not used ?

ahmadshaheer commented:
> * on clicking a log line it opens in the same tab instead of a new one, is there a reason for this because in other places like logs we open it in a new tab.

I had added support for opening the logs explorer in a new tab if we ctrl+click on any log. but it's a better UX to open the log line in a new tab by default. I made the changes :+1: 

> * Sometimes if I click on the logs again for a span, even though the request is successful the data is not rendered in the UI and the loading bar doesn't disappear.

Fixed :raised_hands:

aks07 commented on frontend/src/components/SignozRadioGroup/SignozRadioGroup.tsx:
```diff
@@ -5,7 +5,7 @@ import { RadioChangeEvent } from 'antd/es/radio';
 interface Option {
 	value: string;
-	label: string;
+	label: string | React.ReactNode;
```
why did we remove this? Is this not expected input anymore, now or in the future?

ahmadshaheer replied:
we haven't removed it, @aks07, I have just added another possible value type for label

aks07 commented on frontend/package.json:
```diff
@@ -44,6 +44,7 @@
 		"@sentry/react": "8.41.0",
 		"@sentry/webpack-plugin": "2.22.6",
```
check if this is right. we might be using this elsewhere...should ideally resolve after taking latest pull from main

aks07 commented on frontend/src/container/SpanDetailsDrawer/SpanLogs/SpanLogs.tsx:
```diff
@@ -0,0 +1,277 @@
+	);
+
+	// Footer rendering for pagination
+	const hasReachedEndOfLogs = false;
```
could we have avoided the above 2 useEffects by adding a onSuccess in the useQuery hook?

aks07 replied:
onSuccess: (extractedResults) => {
    // replaces the first useEffect
    handleNewData(extractedResults);
  },
  onSettled: () => {
    // replaces the second useEffect (runs on success or error)
    setIsPaginating(false);
  },

ahmadshaheer commented on frontend/src/container/SpanDetailsDrawer/SpanLogs/useSpanContextLogs.ts:
```diff
@@ -0,0 +1,281 @@
+		const combined = [...afterLogs.reverse(), ...spanLogs, ...beforeLogs];
+		setAllLogs(combined);
+	}, [beforeLogs, spanLogs, afterLogs]);
+
```
as discussed, will take this up separately, @aks07
